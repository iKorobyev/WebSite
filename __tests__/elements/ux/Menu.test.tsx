import React from 'react'
import { cleanup, render, waitFor, act } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Menu } from '~ux'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('Listbox', () => {
  it('renders correctly', async () => {
    const props = {
      show: true,
      options: [{ label: 'Test', value: 'test' }],
      onShow: jest.fn(),
      onHide: jest.fn()
    }

    const { container } = render(<Menu {...props}>Test</Menu>)
    await waitFor(() => container)

    expect(container).toMatchSnapshot()
  })
})
