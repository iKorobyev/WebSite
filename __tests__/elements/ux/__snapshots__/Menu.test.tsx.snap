// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listbox renders correctly 1`] = `
<div>
  <div
    class="relative inline-block text-left undefined"
  >
    <div
      class="inline-flex justify-center cursor-pointer"
    >
      Test
    </div>
    <div
      class="transition ease-out duration-200 transform opacity-0 scale-90"
    >
      <div
        class="w-full max-h-200 overflow-y-scroll absolute left-0 mt-2 origin-top-right bg-white rounded-6 shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none z-1"
        id="headlessui-menu-items-2"
        role="menu"
        tabindex="0"
      >
        <button
          class="group flex rounded-md items-center w-full px-2 py-2 text-14 text-gray-900"
          id="headlessui-menu-item-1"
          role="menuitem"
          tabindex="-1"
        >
          Test
        </button>
      </div>
    </div>
  </div>
</div>
`;
